### ISSUE TITLE
Looking forward to your code

### ISSUE THREAD

**Issue by mobulan** [allenhaozhu/protoLP#1]
Thank you very much for your work, hope to release the code of this project as soon as possible.

**Comment by allenhaozhu** [comment]
> 

Sure, these days I am preparing for the paper works about PhD thesis, so I guess that I would submit codes before 15 June. Thank you for focusing our works.

**Comment by zhuhsingyuu** [comment]
Thank you very much for your work, also hope to release the code of this project as soon as possible.

**Comment by allenhaozhu** [comment]
> Thank you very much for your work, also hope to release the code of this project as soon as possible.

So sorry for my late submission. I just finish my vacation with my mum. I will do this this week.

**Comment by zhuhsingyuu** [comment]
> > Thank you very much for your work, also hope to release the code of this project as soon as possible.
> 
> So sorry for my late submission. I just finish my vacation with my mum. I will do this this week.

Thank you very much!

**Comment by mobulan** [comment]
The currently published code is not complete, especially the codes of the label propagation algorithm, prototype construction, training process, and related parameter settings are missing. This is very important for us to further understand your method. Can you provide as much as possible, thank you very much！

**Comment by allenhaozhu** [comment]
I checked the code. The codes include all things you mentioned, although
two of them are not expected with the unbalance setting and the version
without OT (I will fix this week). So what's your question? Maybe you
misunderstand the code?

Mobulan ***@***.***> 于2023年7月12日周三 19:28写道：

**Comment by mobulan** [comment]
I'm sorry, I misunderstood the code and the content of the text, this code does contain all the content, I have deleted the related issues on github, thank you very much for your reply.
In addition, under my current pytorch 2.0 settings, the accuracy of the reproduction will be about 0.2% lower than that marked in the text, which is the same as iLPC. Do I need to make any additional settings?

从 Windows 版邮件发送

发件人: allenhaozhu
发送时间: 2023年7月13日 20:43
收件人: allenhaozhu/protoLP
抄送: Mobulan; Author
主题: Re: [allenhaozhu/protoLP] Looking forward to your code (Issue #1)

I checked the code. The codes include all things you mentioned, although 
two of them are not expected with the unbalance setting and the version 
without OT (I will fix this week). So what's your question? Maybe you 
misunderstand the code? 

Mobulan ***@***.***> 于2023年7月12日周三 19:28写道：

**Comment by allenhaozhu** [comment]
sorry, I am busy finding a job these days.  About the performance with
Sinkhorn distance, we have mentioned in
https://zhuanlan.zhihu.com/p/633999016. It is a trick for all methods using
Sinkhorn distance, that is why we propose this method. In unbalanced
setting, all these methods will be crashed. The 0.2 ACC gap is from some
dataset-specific tricks such as renormalization anchors with L2
normalization. Thanks for your attention.

Mobulan ***@***.***> 于2023年7月14日周五 00:50写道：

**Comment by mobulan** [comment]
maybe there is a small mistake for the file name? I can re-implement your result with the file “without _full”, maybe this version use the sinkhorn distance发自莫布兰的 iPhone在 2023年7月21日，08:51，allenhaozhu ***@***.***> 写道：﻿
sorry, I am busy finding a job these days.  About the performance with

Sinkhorn distance, we have mentioned in

https://zhuanlan.zhihu.com/p/633999016. It is a trick for all methods using

Sinkhorn distance, that is why we propose this method. In unbalanced

setting, all these methods will be crashed. The 0.2 ACC gap is from some

dataset-specific tricks such as renormalization anchors with L2

normalization. Thanks for your attention.



Mobulan ***@***.***> 于2023年7月14日周五 00:50写道：

**Comment by mobulan** [comment]
in addition, there is an error when use the resnet-12 backbone in "without _full" file. i comment the line 293 and the problem can be resolved (`ndatas[:,] = torch.pow(ndatas[:,]+1e-6, beta)`) and the results can also be reimplemented.
